service/src/main.rs

```diff
@@ -90,18 +141,31 @@ impl Inclusion for InclusionServiceArc {
                         )),
                     }));
                 }
-                JobStatus::Failed(error) => {
+                JobStatus::Failed(error, None) => {
                     return Ok(Response::new(GetKeccakInclusionResponse {
                         status: ResponseStatus::Failed as i32,
-                        response_value: Some(ResponseValue::ErrorMessage(error)),
+                        response_value: Some(ResponseValue::ErrorMessage(format!("{error:?}"))),
                     }));
                 }
-                _ => return Err(Status::internal("Invalid state in proof_tree")),
+                JobStatus::Failed(error, retry_status) => {
+                    // TODO: retry or not?
+                    return Ok(Response::new(GetKeccakInclusionResponse {
+                        status: ResponseStatus::Waiting as i32,
+                        response_value: Some(ResponseValue::StatusMessage(format!(
+                            "Retryring: {retry_status:?} ||| Previous Error: {error:?}"
+                        ))),
+                    }));
+                }
```

Comment:
Unclear if we want to retry when a user hits the RPC (repeatedly), or have some dedicated call to request an retry

I would vote we do add a retryable job back to the queue each time the RPC gets a call for a previously failed job.
```suggestion
                JobStatus::Failed(error, retry_status) => {
                    // send a job back to the worker
                    return Ok(Response::new(GetKeccakInclusionResponse {
                        status: ResponseStatus::Waiting as i32,
                        response_value: Some(ResponseValue::StatusMessage(format!(
                            "Retryring: {retry_status:?} ||| Previous Error: {error:?}"
                        ))),
                    }));
                }
```